### PR TITLE
Add Rubocop Linter to Travis CI

### PR DESCRIPTION
**Pull request by colinsoleim**
### Description

This PR fixes all current rubocop errors and adds `bundle exec rubocop` to the Travis CI script. Most of the changes came from running the prettier and rubocop autoformatters, but I also refactored a few sections that rubocop warned about being overly complex.

I understand if we want to wait until after the MVP is launched to merge this in so people don't have to focus too much on rubocop styles, but the number of rubocop warnings was relatively low, so I decided to push this up now to at least get everyone's opinions.

### Type of change

* Project Refactoring

### How will this affect user permissions?

No Impact.

### How Has This Been Tested?

Extensive local testing and all existing Rspec tests pass.

### CODE REVIEW

**Comment by compwron** [comment]
We had decided during r4g weekend to have rubocop off, and then turn it on later. I do still worry that it will be more inconvenience than convenience for contributors. I'll review the changeset in more detail in a bit. :)

**Comment by colinsoleim** [comment]
> We had decided during r4g weekend to have rubocop off, and then turn it on later. I do still worry that it will be more inconvenience than convenience for contributors. I'll review the changeset in more detail in a bit. :)

No worries, that makes total sense to me. I removed the rubocop command from Travis CI for now!

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 language: ruby
-cache: bundler
+cache:
+  bundler: true
+  directories:
+     - /home/travis/.rvm/
```
Comment (compwron) [inline .travis.yml:5]: cool

File: bin/bundle [hunk 2]
```diff
@@ -14,7 +14,7 @@ m = Module.new do
   module_function
 
   def invoked_as_script?
-    File.expand_path($0) == File.expand_path(__FILE__)
+    File.expand_path($PROGRAM_NAME) == File.expand_path(__FILE__)
```
Comment (compwron) [inline bin/bundle:17]: interesting, will this get wiped out by `rails app:update:bin` ?

File: spec/policies/casa_case_policy/scope_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe CasaCasePolicy::Scope do
```
Comment (compwron) [inline spec/policies/casa_case_policy/scope_spec.rb:3]: :)

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -1,10 +1,7 @@
 require 'spec_helper'
 ENV['RAILS_ENV'] ||= 'test'
-require File.expand_path('../config/environment', __dir__)
-# Prevent database truncation if the environment is production
-if Rails.env.production?
-  abort('The Rails environment is running in production mode!')
-end
+require File.expand_path('../config/environment', __dir__) # Prevent database truncation if the environment is production
```
Comment (compwron) [inline spec/rails_helper.rb:3]: This comment ended up on the wrong line, it should be on the below line. Not a big deal.

**Comment by colinsoleim** [comment]
@compwron thanks for reviewing and merging such a big PR!

**Pull request merged**